src/main/java/InputData/loginInputData.java

```diff
@@ -1,4 +1,31 @@
 package InputData;
 
 public class loginInputData {
-}
+    private final String username;
+    private final String password;
+    private boolean isArtist;
+
+    public loginInputData(String username, String password, boolean isArtist) {
+        this.isArtist = isArtist;
+        this.username = username;
+        this.password = password;
+    }
+    /**
+     * @return the name of user
+     */
+    public String getUsername() {
+        return this.username;
+    }
+    /**
+     * @return the password of user
+     */
+    public String getPassword() {
+        return this.password;
+    }
+    /**
+     * @return whether the user is an artist
+     */
+    public boolean isArtist() {
```

Comment:
RE: my comment in RegisterArtistInputData, I figure this would still be needed in _this_ case (since the two types of users are sharing on this end, it'd be important to distinguish between the two later on).

In RegisterArtistInputData and RegisterInputData, though, they each have their own separate use cases -- so whichever one is being called would implicitly tell us the type.

Reply:
Yes that's why I kept it, I had the exact same train of thought